src/index.rs

```diff
@@ -90,14 +89,16 @@ impl ResourceIndex {
                     let mut parts = entry.split(' ');
 
                     let modified: SystemTime = {
-                        let str = parts.next().ok_or(Error::msg(
+                        let str = parts.next().ok_or(anyhow::Error::msg(
                             "Couldn't find first part of index entry",
                         ))?;
-                        UNIX_EPOCH.add(Duration::from_millis(str.parse()?))
+                        UNIX_EPOCH.add(Duration::from_millis(
+                            str.parse().map_err(|_| ArklibError::Parse)?,
+                        ))
                     };
 
                     let id: ResourceId = {
-                        let str = parts.next().ok_or(Error::msg(
+                        let str = parts.next().ok_or(anyhow::Error::msg(
                             "Couldn't find second part of index entry",
                         ))?;
                         ResourceId::from_str(str)?
```

Comment:
But why `anyhow` here?

Reply:
I fix it using Parse Error seems it's more like a parsing issue